htdocs/compta/facture.php

```diff
@@ -1020,61 +1020,98 @@
+								foreach ($TTotalByTva as $tva => &$total)
+								{
+									$coef = $total / $srcobject->total_ttc; // Calc coef
+									$am = $amount * $coef;
+									$amount_ttc_diff += $am;
+									$amountdeposit[$tva] += $am / (1 + $tva / 100); // Convert into HT for the addline
+								}
 							}
 							else
 							{
-								$amountdeposit = 0;
-
-								if ($result > 0)
+								if ($typeamount == 'amount')
 								{
-									$totalamount = 0;
-									$lines = $srcobject->lines;
-									$numlines=count($lines);
-									for ($i=0; $i<$numlines; $i++)
+									$amountdeposit[] = $valuedeposit;
+								}
+								else
+								{
+									if ($result > 0)
 									{
-										$qualified=1;
-										if (empty($lines[$i]->qty)) $qualified=0;	// We discard qty=0, it is an option
-										if (! empty($lines[$i]->special_code)) $qualified=0;	// We discard special_code (frais port, ecotaxe, option, ...)
-										if ($qualified) $totalamount += $lines[$i]->total_ht;
-									}
+										$totalamount = 0;
+										$lines = $srcobject->lines;
+										$numlines=count($lines);
+										for ($i=0; $i<$numlines; $i++)
+										{
+											$qualified=1;
+											if (empty($lines[$i]->qty)) $qualified=0;	// We discard qty=0, it is an option
+											if (! empty($lines[$i]->special_code)) $qualified=0;	// We discard special_code (frais port, ecotaxe, option, ...)
+											if ($qualified) $totalamount += $lines[$i]->total_ht; // Fixme : is it not for the customer ? Shouldn't we take total_ttc ?
+										}
 
-									if ($totalamount != 0) {
-										$amountdeposit = ($totalamount * $valuedeposit) / 100;
+										if ($totalamount != 0) {
+											$amountdeposit[] = ($totalamount * $valuedeposit) / 100;
+										}
+									} else {
+										setEventMessages($srcobject->error, $srcobject->errors, 'errors');
+										$error ++;
 									}
-								} else {
-									setEventMessages($srcobject->error, $srcobject->errors, 'errors');
-									$error ++;
 								}
+								
+								$amount_ttc_diff = $amountdeposit[0];
 							}
-
-							$result = $object->addline(
-									$langs->trans('Deposit'),
-									$amountdeposit,		 	// subprice
-									1, 						// quantity
-									$lines[$i]->tva_tx,     // vat rate
-							        0,                      // localtax1_tx
-									0, 						// localtax2_tx
-									(empty($conf->global->INVOICE_PRODUCTID_DEPOSIT)?0:$conf->global->INVOICE_PRODUCTID_DEPOSIT), 	// fk_product
-									0, 						// remise_percent
-									0, 						// date_start
-									0, 						// date_end
-									0, $lines[$i]->info_bits, // info_bits
-									0, 						// info_bits
-									'HT',
-									0,
-									0, 						// product_type
-									1,
-									$lines[$i]->special_code,
-									$object->origin,
-									0,
-									0,
-									0,
-									0,
-									$langs->trans('Deposit')
-								);
+							
+							foreach ($amountdeposit as $tva => $amount)
```

Comment:
If new option is set, the key of array amountdeposit is tva. OK
But with standard behaviour (without option, line 1066, the key is not defined:   $amountdeposit[] = ... where previous code was using $lines[$i]->tva_tx.